interactions/client/models/command.py

```diff
@@ -181,3 +185,245 @@ class ApplicationCommand(DictSerializerMixin):
     name_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
     description_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
+
+
+class StopCommand:
+    """
+    A class that when returned from a command, the command chain is stopped.
+
+    Usage:
+    ```py
+    @bot.command()
+    async def foo(ctx):
+        ... # do something
+        return StopCommand  # does not execute `bar`
+        # or `return StopCommand()`
+
+    @foo.subcommand()
+    async def bar(ctx): ...
+    ```
+    """  # TODO: change docstring
+
+
+@define()
+class BaseResult(DictSerializerMixin):
+    """docstring"""  # TODO: change docstring
+
+    result: Any = field()
+
+    def __call__(self) -> Any:
+        return self.result
+
+
+@define()
+class GroupResult(DictSerializerMixin):
+    """docstring"""  # TODO: change docstring
+
+    result: Any = field()
+    parent: BaseResult = field()
+
+    def __call__(self) -> Any:
+        return self.result
+
+
+@define()
+class Command(DictSerializerMixin):
+    """docstring"""  # TODO: change docstring
+
+    client: Client = field()
+    coro: Callable[..., Awaitable] = field()
+    type: ApplicationCommandType = field(converter=ApplicationCommandType)
+    base: str = field()
+    description: str = field()
+    options: Optional[List[Option]] = field(converter=convert_list(Option), default=None)
+    scope: List[int] = field(converter=convert_list(int))
+    version: int = field(default=None)
+    default_member_permissions: str = field()
+    dm_permission: bool = field(default=None)
+    name_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
+    description_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
+
+    def __attrs_post_init__(self) -> None:
+        self.coroutines: Dict[str, Callable[..., Awaitable]] = {}
+        if self.base in (MISSING, None):
+            self.base = self.coro.__name__
+        if self.description in (MISSING, None) and self.type == ApplicationCommandType.CHAT_INPUT:
+            self.description = getdoc(self.coro) or "No description set"
+            self.description = self.description.split("\n", 1)[0]
+
+    def __call__(self, *args, **kwargs) -> Awaitable:
+        return self.coro(*args, **kwargs)
+
+    @property
+    def full_data(self) -> Union[dict, List[dict]]:
+        """Returns the command in JSON format."""  # TODO: change docstring
+        data = command(
+            type=self.type,
+            name=self.base,
+            description=self.description if self.type == 1 else None,
+            options=self.options,
+            scope=self.scope,
+            name_localizations=self.name_localizations,
+            description_localizations=self.description_localizations,
+            default_member_permissions=self.default_member_permissions,
+            dm_permission=self.dm_permission,
+        )
+        return data if isinstance(data, list) else [data]
```

Comment:
If you want to use this for sync it has not to be a list for global commands iirc

Reply:
Yes, I know that, I just did it for consistency and the potential to just do `.extend(data)` later on if possible.
I will tackle sync later on.

Reply:
I just made it `return command(...)`, thanks